int32[] path_indexes